pkg/registry/core/service/storage/storage.go

```diff
@@ -131,42 +131,36 @@ func (r *StatusREST) Update(ctx context.Context, name string, objInfo rest.Updat
 // applies on Get, Create, and Update, but we need to distinguish between them.
 //
 // This will be called on both Service and ServiceList types.
-func (r *GenericREST) defaultOnRead(obj runtime.Object) error {
+func (r *GenericREST) defaultOnRead(obj runtime.Object) {
 	service, ok := obj.(*api.Service)
 	if ok {
-		return r.defaultOnReadService(service)
+		r.defaultOnReadService(service)
```

Comment:
this still works, but it's a little weird not to return. You do still want to return in this branch, right? 

maybe a type switch would be more normal

Reply:
ACK, type switch would be.